### DISCUSSION TITLE
Problems:  • Device cannot be used while the lid is closed

### DISCUSSION THREAD

**Discussion by kunzaatko** [fwupd/fwupd#4926] [Q&A] [answered]
version: _runtime   org.freedesktop.fwupd         1.8.3_

When I run `fwupdmgr get-devices` I get under the _UEFI Device Firmware:_ device the problem in the title. When I run `fwupdmgr update` it fails with: _System Firmware is not currently updatable: Device cannot be used while the lid is closed_. First I thought that it was the same as in issue #4870. But this issue is closed with the problem being stated as resolved... What am I doing wrong and how to go deal with this problem?

**Comment by superm1** [comment]
Open up the lid and run the update.

**Reply by kunzaatko** [reply]
Yeah, well I though this was the intent of the message. But my lid is open. That is why I thought that I misunderstood somehow.
If this is correct, than I guess I will make this an issue instead of a discussion, do you agree @superm1?

**Reply by superm1** [reply]
Oh interesting.  So perhaps your lid switch sensor is reporting the wrong value then?  Does it let you run the command with lid closed?  The update might fail with it closed, but that would at least confirm that your sensor is functioning but inverted.

**Reply by kunzaatko** [reply]
I can confirm that when using an external monitor with the lid closed, the command gives the same output. To address the lid sensor... The command `bat /proc/acpi/lid/LID/state` returns _state: open_ as expected and with the external monitor and closed lid, it returns _state: closed_ also as expected...

**Reply by superm1** [reply]
Can you check if it's happening at the upower level using d-bus?  We pretty much are supposed to be mirroring that.
https://github.com/fwupd/fwupd/blob/fb701a0426c643b93fecaa4270d3e6cda7f073d8/plugins/upower/fu-plugin-upower.c#L99

**Reply by kunzaatko** [reply]
Yes, so I think you are up to something with this. When I dump the state with `upower -d` this is an exert from the output:
> Daemon:
>    daemon-version:  0.99.20
>    on-battery:      no
>    lid-is-closed:   yes
>    lid-is-present:  yes
>    critical-action: PowerOff

So this is rather an issue coming from `upower` then... What would you suggest would be the best route to go with this information? Should it be passed as an issue with `fwupd` as well or not? Thank you for your help!

**Reply by superm1** [reply]
Let's figure out upower issue before we decide what to do with fwupd.  If fwupd can't trust upower, we shouldn't be using it :)

**Reply by superm1** [reply]
https://gitlab.freedesktop.org/upower/upower/-/issues/174 Comes to mind, but I don't think that should be it, you're on 0.99.20 which is just a month old.  I'd say open a new issue upstream there.

**Reply by kunzaatko** [reply]
I submitted an [issue](https://gitlab.freedesktop.org/upower/upower/-/issues/202) to `upower` and will keep you posted. :)

**Reply by hughsie** [reply]
Great, thanks. Well done on the debugging.

**Comment by kunzaatko** [comment, marked as answer]
Hi again! I am here to inform you about the conclusion of this problem. In short this is probably not a bug of `upower` itself. On my system there where two devices that where marked as a _lid switch_ and the first one does not correspond with the lid switch. If anyone would want further details, they can be found in the `upower` issue here [upower/upower#202](https://gitlab.freedesktop.org/upower/upower/-/issues/202). This could be a kernel bug and I will be following up on that. Thank you all for your suggestions and the help with debugging this!
